### DISCUSSION TITLE
What risk poses isar 4?

### DISCUSSION THREAD

**Discussion by frederikstonge** [isar-community/isar#67] [General]
So I started making an app and decided to go with isar 4 because it supported freezed models. I didn't really notice that it wasn't maintained anymore, and I thought that my development process would be slower than the actual release of isar 4. 

Now, I'm at the verge of releasing my app. I have switched to the community version (thank you guys) because there are at least some fixes and maintenance. However, there is still this big warning that we should not use isar 4 in production. What risk am I taking by using isar 4 in production?

Is there a possibility to downgrade to isar 3 and still use freezed?

Thank you

**Comment by vicenterusso** [comment]
I use freezed with isar v3, but not directly in Collections. But I used [dart_mappable](https://pub.dev/packages/dart_mappable) on another project directly on collections, iirc.

**Reply by vicenterusso** [reply]
Have you tried this alternative @frederikstonge ?

**Comment by definitelyme** [comment]
@frederikstonge @vicenterusso I have a similar issue. On the first paragraph of the plugin's [CHANGELOG](https://pub.dev/packages/isar/versions/4.0.0-dev.14/changelog) for v4.0.0, it says `This version does not support database migration yet and cannot open Isar v3 databases.`

My question is, how did you handle database migration in your project? Does the community fork support this?

**Reply by vicenterusso** [reply]
Hello, sorry the late response. At the moment v4 cannot fully migate from v3. My best quest would be sync everything to cloud and pull back on v4, or some sort of export directly on device (json, etc)
